src/nix/profile.cc

```diff
@@ -329,7 +335,63 @@ struct CmdProfileInstall : InstallablesCommand, MixDefaultProfile
             manifest.elements.push_back(std::move(element));
         }
 
-        updateProfile(manifest.build(store));
+        try {
+            updateProfile(manifest.build(store));
+        } catch (BuildProfileConflictError & conflictError) {
+            std::string fallbackInstallable = "INSTALLABLE";
+            std::string conflictingInstallable = [&]() {
+                auto maybeConflictingStorePath = StorePath::maybeParseAbsolutePath(conflictError.conflictingPkgPath);
+                if (!maybeConflictingStorePath) {
+                    printError("warning: unable to determine installable from conflicting package path (%1%)", conflictError.conflictingPkgPath);
+                    return fallbackInstallable;
+                }
+                auto conflictingStorePath = *maybeConflictingStorePath;
+                auto conflictingManifestElement = find_if(manifest.elements.begin(), manifest.elements.end(), [&](const ProfileElement & currentProfileElement) {
+                    return currentProfileElement.hasStorePath(conflictingStorePath);
+                });
+                if (conflictingManifestElement == manifest.elements.end()) {
+                    printError("warning: unable to find profile element for conflicting store path (%1%)", conflictingStorePath.to_string());
+                    return fallbackInstallable;
+                }
+                if (!conflictingManifestElement->source) {
+                    printError("warning: conflicting store path does not have a source (%1%)", conflictingStorePath.to_string());
+                    return fallbackInstallable;
+                }
+                return conflictingManifestElement->source->originalRef.to_string();
+            }();
+
+            throw Error(
+                "There is an existing package that has a conflicting file:\n"
+                "\n"
+                "  %1%\n"
+                "\n"
+                "The conflicting file:\n"
+                "\n"
+                "  %3%\n"
```

Comment:
This reads confusing to me, as if the second store path is contained in the first.

```suggestion
                "An existing package already symlinks the following file to the user's profile:\n"
                "\n"
                "  %1%\n"
                "\n"
                "The conflicting file:\n"
                "\n"
                "  %3%\n"
```

Reply:
Indeed, it's worded a bit convoluted. I'd like to avoid mentioning symlinking to the user. They don't necessarily know profiles are build from symlinks. I reworded it as:

> An existing package already provides the following file to the user's profile